pkg/registry/apps/deployment/storage/storage.go

```diff
@@ -404,5 +418,12 @@ func (i *scaleUpdatedObjectInfo) UpdatedObject(ctx context.Context, oldObj runti
 	// move replicas/resourceVersion fields to object and return
 	deployment.Spec.Replicas = scale.Spec.Replicas
 	deployment.ResourceVersion = scale.ResourceVersion
+
+	updatedEntries, err := fieldmanager.MergeSubresourceEntries(deployment.ManagedFields, scale.ManagedFields, "apps/v1", "spec", "replicas")
```

Comment:
rather than assuming `apps/v1` and `spec.replicas`, this should switch on the group/version of the request (defaulting to apps/v1 is ok for calls that don't have request info), something like:

https://github.com/kubernetes/kubernetes/blob/f4250198eac34e1adae7c627b9cb0cb4252bdb1e/pkg/registry/apps/deployment/strategy.go#L52-L56

If request info is found, but the group/version is not recognized, that should be a failure so we don't accidentally import apps/v1 fieldpaths into future versions automatically/incorrectly

Reply:
ideally this could be encapsulated into a form where every scale subresource storage had a constant mapping of request group/version to fieldpath mapping, something like 

```go
mappings := map[schema.GroupVersion]mapping{
  schema.GroupVersion{Group:"apps",Version:"v1"}: mapping{
    From:[]string{"spec","replicas"}, // scale field path
    To: []string{"spec","replicas"}, // deployment field path
  },
}
```

or 

```go
mappings := map[schema.GroupVersion]mapping{
  schema.GroupVersion{Group:"acme.com",Version:"v1"}: mapping{
    From:[]string{"spec","replicas"}, // scale field path
    To: []string{"some","fieldPath","to","replicas"}, // v1 custom resource type field path
  },
  schema.GroupVersion{Group:"acme.com",Version:"v2"}: mapping{
    From:[]string{"spec","replicas"}, // scale field path
    To: []string{"some","other","fieldPath","to","replicas"}, // v2 custom resource type field path
  },
}
```

then select a mapping based on the request version, then call a helper, passing in the managed fields and mapping

Reply:
Thank you so much, makes sense to me.

I definitely like the idea of keeping a` mapping`, if I understand correctly the mapping is not constant right? because custom resource could define their own replicas path. In other words, there should be a way to "add" items to the mapping. Is that what you were thinking?

Reply:
I don't know about the mapping to `.spec.replicas` in the scale object? That seems to be pretty fix, and doesn't depend on the type we're converting to/from. I think it's fine to have a map of type (Deployment, etc...) to replicas field.

Reply:
I addressed this in https://github.com/kubernetes/kubernetes/pull/98377/commits/79cdc00f8b65c46187a374d92cf8affccbebefb2, there is probably more refactoring we can do, but would be great if you could double-check that I am going in the right direction.